third-party/realdds/src/dds-stream.cpp

```diff
@@ -0,0 +1,82 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-stream.h>
+#include "dds-stream-impl.h"
+
+#include <stdexcept>
+
+namespace realdds {
+
+
+dds_video_stream::dds_video_stream( int8_t type, const std::string & group_name )
+    : _impl( std::make_shared< dds_video_stream::impl >( type, group_name ) )
+{
+}
+
+int8_t dds_video_stream::get_type() const
+{
+    return _impl->_type;
+}
+
+const std::string & dds_video_stream::get_group_name() const
+{
+    return _impl->_group_name;
+}
+
+void dds_video_stream::add_profile( const dds_stream::profile & prof, bool default_profile )
+{
+    if ( _impl->_type != prof.type )
+    {
+        throw std::runtime_error( "profile of different type then stream" );
```

Comment:
Maybe better to add the type received and the internal type to the log ? Your choice..

Reply:
Done